src/Umbraco.Web/PublishedCache/NuCache/DataSource/MsgPackContentNestedDataSerializer.cs

```diff
@@ -48,10 +53,42 @@ public ContentNestedData Deserialize(string data)
 
         public string Serialize(ContentNestedData nestedData)
         {
+            Optimize(nestedData);
+
             var bin = MessagePackSerializer.Serialize(nestedData, _options);
             return Convert.ToBase64String(bin);
         }
 
+        /// <summary>
+        /// Compress properties and map property names to shorter names
+        /// </summary>
+        /// <param name="nestedData"></param>
+        private void Optimize(ContentNestedData nestedData)
+        {
+            if (_propertyOptions.PropertyMap != null && _propertyOptions.PropertyMap.Any())
+            {
+                foreach (var map in _propertyOptions.PropertyMap)
+                {
+                    if (map.Value.compress.Equals(NucachePropertyCompressionLevel.SQLDatabase))
```

Comment:
maybe this should also do the nucache option as well

Reply:
My comments on this have been way off. There are two enums.
    public enum NucachePropertyCompressionLevel
    {
        None = 0, //No property compression
        SQLDatabase = 1, //Compress the property before being written to the database
        NuCacheDatabase = 2 // Compress the property when writing to nucache bplustree after reading from the database
    }
    /// <summary>
    /// If/where to decompress custom properties for nucache
    /// </summary>
    public enum NucachePropertyDecompressionLevel
    {
        NotCompressed = 0, // No decompression required
        Immediate = 1, //Decompress the property after it's read from the database (at boot or rebuild)
        Lazy = 2 //Decompress the property when .ToString() is called.
    }

Reply:
Yep all good, have figured it out :) I've removed the SQLDatabase  option for now along with the 'MsgPackContentNestedDataSerializer.Optimize'  since I don't really think anyone would choose to not use the binary storage format since it's much faster and smaller. We can add back later if we think it's important.

Reply:
The SQLDatabase option is still binary, and my preferred option for my scenario. The goal with it was to save on compression and memory on boot as it does not need to hold both the string value and the pickled byte[] at boot. I updated further explanation as a comment on the main thread instead of here (my apologies, it's midnight here).